lib/glific/contacts/import.ex

```diff
@@ -115,6 +134,32 @@ defmodule Glific.Contacts.Import do
     end
   end
 
+  ## create contact fields from sheet header and the first contact
+  @spec create_contact_fields(list) :: list
+  defp create_contact_fields(contacts) do
+    first_contact = Enum.at(contacts, 0)
+
+    Enum.each(first_contact.contact_fields, fn {field, _value} ->
+      field = Glific.string_snake_case(field)
+
+      case Repo.get_by(Glific.Contacts.ContactsField, %{shortcode: field},
+             organization_id: first_contact.organization_id
+           ) do
+        nil ->
+          ContactField.create_contact_field(%{
```

Comment:
let's create a function called may_be_create_contact_fields function and then we can use upsert there.

Reply:
Done